pkg/kubelet/cm/cpumanager/policy_static.go

```diff
@@ -342,9 +348,36 @@ func (p *staticPolicy) Allocate(s state.State, pod *v1.Pod, container *v1.Contai
 		}
 	}
 	if cpuset, ok := s.GetCPUSet(string(pod.UID), container.Name); ok {
-		p.updateCPUsToReuse(pod, container, cpuset)
-		klog.InfoS("Static policy: container already present in state, skipping", "pod", klog.KObj(pod), "containerName", container.Name)
-		return nil
+		if utilfeature.DefaultFeatureGate.Enabled(features.InPlacePodVerticalScaling) {
+			klog.InfoS("Static policy: container already present in state, attempting InPlacePodVerticalScaling", "pod", klog.KObj(pod), "containerName", container.Name)
```

Comment:
uhm, the CPU allocation is meant to be performed at admission time. Does this mean in-place VPA will always attempt a new admission loop?

Reply:
Yes, so corner cases need to be handled. Thanks for your comment.

Reply:
> uhm, the CPU allocation is meant to be performed at admission time. Does this mean in-place VPA will always attempt a new admission loop?

Answer is yes, it is handled inside SyncPod code under handlePodResourcesResize, please check https://github.com/kubernetes/kubernetes/blob/657c2f9f793d64405e4a3fddd440b6d70dd70256/pkg/kubelet/kubelet.go#L1949
and
https://github.com/kubernetes/kubernetes/blob/657c2f9f793d64405e4a3fddd440b6d70dd70256/pkg/kubelet/kubelet.go#L2801

Reply:
This deserve more investigation, at least from me - not familiar with VPA. I'm worried about races between admission checks and VPA attempts. Up until now the allocation logic was run only at admission time, where it is strongly and implicitely serialized.

Reply:
https://github.com/kubernetes/kubernetes/pull/124296 , must be merged first to be able to create in this PR  new add on test cases which will help get answers on your concerns. Unit tests written in this PR are not sufficient for testing race conditions between admission checks and VPA attempts, i believe. What do you think?

Reply:
@ffromani glad to share that test/e2e_node/pod_resize_test.go now (with latest patch) test alongside CPU Manager Static Policy without any Policy Options with integer CPU requests and with FullPCPUsOnlyOption :-) to build us more confidence.

Reply:
Will update back soon, doing a refactoring as result of https://github.com/kubernetes/kubernetes/pull/123319#issuecomment-2359919840